cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -436,22 +436,11 @@ void MapNode::SubmitTask(std::function<Result<ExecBatch>(ExecBatch)> map_fn,
     return Status::OK();
   };
 
-  if (executor_) {
-    status = task_group_.AddTask([this, task]() -> Result<Future<>> {
-      return this->executor_->Submit(this->stop_source_.token(), [this, task]() {
-        auto status = task();
-        if (this->input_counter_.Increment()) {
-          this->Finish(status);
-        }
-        return status;
-      });
-    });
-  } else {
-    status = task();
-    if (input_counter_.Increment()) {
-      this->Finish(status);
-    }
+  status = task();
+  if (input_counter_.Increment()) {
+    this->Finish(status);
   }
+
   // If we get a cancelled status from AddTask it means this node was stopped
   // or errored out already so we can just drop the task.
   if (!status.ok() && !status.IsCancelled()) {
```

Comment:
We don't need the comment (or the check for `!status.IsCancelled`) anymore.

Reply:
Made the change.